rust/index/src/hnsw_provider.rs

```diff
@@ -162,78 +164,49 @@ impl HnswIndexProvider {
         source_id: &Uuid,
         index_storage_path: &Path,
     ) -> Result<(), Box<HnswIndexProviderFileError>> {
-        // Fetch the files from storage and put them in the index storage path
+        // Fetch the files from storage and put them in the index storage path.
+        // TODO: Fetch multiple chunks in parallel from S3.
```

Comment:
curious at what level you think this should occur (We can always munge things in future, not saying any changes should happen now, just curious)